pilot/pkg/model/jwks_resolver.go

```diff
@@ -260,6 +260,11 @@ func (r *JwksResolver) resolveJwksURIUsingOpenID(issuer string) (string, error)
 		return uri.(string), nil
 	}
 
+	// Default to https if the provided url has no scheme
```

Comment:
please do not change the resolveJwksURIUsingOpenID() for doing anything not related to the jwks uri resolution of OpenID. And it will actually also not work because I just realized that the function is only called for alpha policy in SetAuthenticationPolicyJwksURIs(), but that's another issue that we should fix in a separate PR. (cc @diemtvu )

Before the above issue is fixed, I think it's better to go back to add this in `getRemoteContentWithRetry()` as your original PR, sorry for the back and forth here, also I think you have figured out how to add the unit test, that should also be included.

Thank you!